tstime/rate/rate_test.go

```diff
@@ -185,7 +185,7 @@ func TestLongRunningQPS(t *testing.T) {
 	// This will still offer ~500 requests per second,
 	// but won't consume outrageous amount of CPU.
 	start := time.Now()
-	end := start.Add(5 * time.Second)
+	end := start.Add(1 * time.Second)
```

Comment:
Doesn't this increase the likelihood of a flake on a highly loaded machine?

Reply:
How about trying it for 1 second up to 5 tries?

Reply:
It is not obvious to me the impact. It might actually make it more reliable, if the failures occur when a single long pause happens--then there's a smaller window to get unlucky with the scheduler. I'm inclined to keep it shorter for now, and get fancy with retries if/when there's an actual problem. It went from >99% failure to 0% failure in my tests here.